src/Datadog.Trace.ClrProfiler.Managed/Emit/DynamicMethodBuilder.cs

```diff
@@ -81,23 +86,39 @@ internal static class DynamicMethodBuilder<TDelegate>
                     }
                 }
 
-                if (genericTypes.Length > 0 && !methods[i].ContainsGenericParameters)
+                var candidateGenericTypes = methods[i].GetGenericArguments();
```

Comment:
I like this change, I think GetGenericArguments is the right method to call here